frontend/pages/hosts/HostDetailsPage/SoftwareListRow/SoftwareListRow.jsx

```diff
@@ -35,8 +36,23 @@ class SoftwareListRow extends Component {
 
     const type = TYPE_CONVERSION[source] || "Unknown";
 
+    const vulnerabilitiesIcon = () => {
+      if (vulnerabilities && vulnerabilities.length === 1) {
+        return <IconToolTip text={"1 vulnerability detected"} issue isHtml />;
+      } else if (vulnerabilities) {
+        return (
+          <IconToolTip
+            text={`${vulnerabilities.length} vulnerabilities detected`}
+            issue
+            isHtml
+          />
+        );
+      }
+    };
```

Comment:
Assuming `vulnerabilities` is an array...
```suggestion
    const vulnerabilitiesIcon = () => {
      if (vulnerabilities.length === 0) {
        return null;
      }
      
      const vulText = vulnerabilities.length === 1 ? "vulnerability" : "vulnerabilities";
      return (
          <IconToolTip
            text={`${vulnerabilities.length} ${vulText} detected`}
            issue
            isHtml
          />
        );
    };
```

Reply:
This way, we follow the pattern of returning a JSX element or null, which will be helpful once we add typescript.